tests/components/sharkiq/test_shark_iq.py

```diff
@@ -0,0 +1,283 @@
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac])
+    shark = SharkVacuumEntity(shark_vac, coordinator)
+
+    # These come from the setup
+    assert isinstance(shark.is_docked, bool) and not shark.is_docked
+    assert (
+        isinstance(shark.recharging_to_resume, bool) and not shark.recharging_to_resume
+    )
+    # Go through the operation modes while it's "off the dock"
+    await shark.async_start()
+    assert shark.operating_mode == shark.state == STATE_CLEANING
+    await shark.async_pause()
+    assert shark.operating_mode == shark.state == STATE_PAUSED
+    await shark.async_stop()
+    assert shark.operating_mode == shark.state == STATE_IDLE
+    await shark.async_return_to_base()
+    assert shark.operating_mode == shark.state == STATE_RETURNING
+
+    # Test the docked modes
+    await shark.async_stop()
+    shark.sharkiq.set_property_value(Properties.RECHARGING_TO_RESUME, 1)
+    shark.sharkiq.set_property_value(Properties.DOCKED_STATUS, 1)
+    assert isinstance(shark.is_docked, bool) and shark.is_docked
+    assert isinstance(shark.recharging_to_resume, bool) and shark.recharging_to_resume
+    assert shark.state == STATE_RECHARGING_TO_RESUME
+
+    shark.sharkiq.set_property_value(Properties.RECHARGING_TO_RESUME, 0)
+    assert shark.state == STATE_DOCKED
+
+    await shark.async_set_fan_speed("Eco")
+    assert shark.fan_speed == "Eco"
+    await shark.async_set_fan_speed("Max")
+    assert shark.fan_speed == "Max"
+    await shark.async_set_fan_speed("Normal")
+    assert shark.fan_speed == "Normal"
+
+    assert set(shark.fan_speed_list) == {"Normal", "Max", "Eco"}
+
+
+@patch.object(SharkIqVacuum, "set_property_value", new=_set_property)
+async def test_shark_vac_properties(hass: HomeAssistant) -> None:
+    """Test all of the shark vacuum property accessors."""
+    ayla_api = MockAyla()
+    shark_vac = _get_mock_shark_vac(ayla_api)
+    coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac])
+    shark = SharkVacuumEntity(shark_vac, coordinator)
+
+    assert shark.name == "Sharknado"
+    assert shark.serial_number == "AC000Wxxxxxxxxx"
+    assert shark.model == "RV1000A"
+
+    assert shark.battery_level == 50
+    assert shark.fan_speed == "Eco"
+    shark.sharkiq.set_property_value(Properties.POWER_MODE, 0)
+    assert shark.fan_speed == "Normal"
+    assert isinstance(shark.recharge_resume, bool) and shark.recharge_resume
+    assert isinstance(shark.low_light, bool) and not shark.low_light
+
+    target_state_attributes = {
+        ATTR_ERROR_CODE: 7,
+        ATTR_ERROR_MSG: "Cliff sensor is blocked",
+        ATTR_RECHARGE_RESUME: True,
+        ATTR_LOW_LIGHT: False,
+    }
+    state_json = json.dumps(shark.device_state_attributes, sort_keys=True)
+    target_json = json.dumps(target_state_attributes, sort_keys=True)
+    assert state_json == target_json
+
+    assert not shark.should_poll
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)